### PR TITLE
Add an option to the workflow config to force use of compat mode

### PR DESCRIPTION
**Pull request by wxtim**
Works with https://github.com/cylc/cylc-rose/pull/322 to close https://github.com/cylc/cylc-rose/issues/319.

Functional test provided by https://github.com/metomi/rose/pull/2779

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` (and `conda-environment.yml` if present).
- [x] Tests are included (or explain why tests are not needed).
- [x] This small change does not require documentation
- [x] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.

### CODE REVIEW

File: cylc/flow/config.py [hunk 1]
```diff
@@ -876,7 +880,7 @@ def _check_implicit_tasks(self) -> None:
             )
         # Allow implicit tasks in back-compat mode unless rose-suite.conf
         # present (to maintain compat with Rose 2019)
-        elif not (self.fpath.parent / "rose-suite.conf").is_file():
+        elif not (self.fdir / "rose-suite.conf").is_file():
```
Comment (oliver-sanders) [inline cylc/flow/config.py:883]: Isn't `self.fdir` a `str`?

Comment (wxtim) [reply cylc/flow/config.py]: Yes. Reverted.

File: cylc/flow/config.py [hunk 2]
```diff
@@ -226,8 +227,11 @@ def __init__(
             workflow: workflow ID
             fpath: workflow config file path
             options: CLI options
+            force_compat_mode:
+                Override compatibility mode checks.
+                https://github.com/cylc/cylc-rose/issues/319
```
Comment (oliver-sanders) [inline cylc/flow/config.py:232]:
```suggestion
            force_compat_mode:
                If True, forces Cylc to use compatibility mode
                overriding compatibility mode checks.
                See https://github.com/cylc/cylc-rose/issues/319

```

File: cylc/flow/workflow_files.py [hunk 3]
```diff
@@ -779,7 +779,7 @@ def get_workflow_title(id_):
     return title
 
 
-def check_deprecation(path, warn=True):
+def check_deprecation(path, warn=True, force_compat_mode=False):
     """Warn and turn on back-compat flag if Cylc 7 suite.rc detected.
 
     Path can point to config file or parent directory (i.e. workflow name).
```
Comment (oliver-sanders) [inline cylc/flow/workflow_files.py:785]:
```suggestion
    Path can point to config file or parent directory (i.e. workflow name).

    Args:
        warn:
            If True, then a warning will be logged when compatibility
            mode is activated.
        force_compat_mode:
            If True, forces Cylc to use compatibility mode
            overriding compatibility mode checks.
            See https://github.com/cylc/cylc-rose/issues/319
```

**Review by oliver-sanders — APPROVED** [review]

File: cylc/flow/config.py [hunk 4]
```diff
@@ -217,7 +217,8 @@ def __init__(
         run_dir: Optional[str] = None,
         log_dir: Optional[str] = None,
         work_dir: Optional[str] = None,
-        share_dir: Optional[str] = None
+        share_dir: Optional[str] = None,
+        force_compat_mode: Optional[bool] = False,
```
Comment (MetRonnie) [inline cylc/flow/config.py:221]:
```suggestion
        force_compat_mode: bool = False,
```

**Review by MetRonnie — APPROVED** [review]

**Pull request merged**